src/ServiceJobList.c

```diff
@@ -0,0 +1,143 @@
+//ServiceJobList.c
+
+// SPDX-FileCopyrightText: 2024, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2024, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+rcl_service_t g_serviceListJobs;
+
+ServiceListJobs_Messages g_messages_ListJobs;
+
+typedef motoros2_interfaces__srv__ListJobs_Response ListJobsResponse;
+
+
+static const size_t RESP_NAMES_LIST_INITIAL_SIZE = 64;
```

Comment:
I don't expect MotoROS2 users to have too many regular INFORM jobs, so this is probably as good a default/starting value as any, but we could obviously go with a higher number.

Reply:
Personally, I think it's better to init to 1, then resize based on the value returned by `mpGetFileCount`. I don't see benefit to preallocating 64.

Reply:
> I assume the maximum nr of jobs on a single controller is actually quite high?

Yeah.  The spec is for 200,000 motion steps and 10,000 non-motion instructions. I don't imagine someone have a 100,000 two-step jobs... but the possibility exists.
(I have seen people run out of job space before.)

Reply:
> Personally, I think it's better to init to 1, then resize based on the value returned by `mpGetFileCount`. I don't see benefit to preallocating 64.

less runtime allocations?

> > I assume the maximum nr of jobs on a single controller is actually quite high?
> 
> Yeah. The spec is for 200,000 motion steps and 10,000 non-motion instructions. I don't imagine someone have a 100,000 two-step jobs... but the possibility exists. (I have seen people run out of job space before.)

So the limits are for steps in a job, not job files themselves? Or am I misinterpreting this?

Reply:
> So the limits are for steps in a job, not job files themselves

Correct. 

There may be another limit for qty of files, due to the header overhead. But I can't find it anywhere.